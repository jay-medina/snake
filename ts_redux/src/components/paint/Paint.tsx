import React from 'react';
import { Board } from '../board/Board';
import { ScoreBoard } from '../scoreboard/ScoreBoard';

export const Paint = () => (
  <div className="snake__app">
    <ScoreBoard highScore={10} score={10} />
    <Board row={25} col={25} />
  </div>
);
