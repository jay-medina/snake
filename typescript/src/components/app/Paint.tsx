import React from 'react';
import ScoreBoard from '../scoreboard/Scoreboard';
import Board from '../board/Board';
import { Snake, Apple } from '../common/types';

export interface AppProps {
  row: number;
  col: number;
  score: number;
  highScore: number;
  snake: Snake;
  apple: Apple;
}

const App: React.StatelessComponent<AppProps> = ({ row, col, score, highScore, snake, apple }) => (
  <div className="snake__app">
    <ScoreBoard score={score} highScore={highScore} />
    <Board apple={apple} snake={snake} row={row} col={col} />
  </div>
);

export default App;
