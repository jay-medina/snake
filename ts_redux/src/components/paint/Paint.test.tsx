import { shallow } from 'enzyme';
import React from 'react';
import { Paint } from './Paint';

describe('<Paint />', () => {
  it('renders the component', () => {
    const component = shallow(<Paint />);
    expect(component).toMatchSnapshot();
  });
});
