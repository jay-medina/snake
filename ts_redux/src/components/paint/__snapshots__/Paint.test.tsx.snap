// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paint /> renders the component 1`] = `
<div
  className="snake__app"
>
  <Scoreboard
    highScore={10}
    score={10}
  />
  <Board
    col={25}
    row={25}
  />
</div>
`;
